### PR TITLE
Add basic auth to github-add-comment

### PR DESCRIPTION
**Pull request by frerikandriessen**
# Changes

Adds the option to use Basic Auth, similarl to how it was done in github-open-pr: https://github.com/tektoncd/catalog/blob/main/task/github-open-pr/0.2/github-open-pr.yaml

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Review by vinamra28 — APPROVED** [review]

File: task/github-add-comment/0.6/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Add a comment to an issue or a pull request
+
+The `github-add-comment` task let you add a comment to a pull request or an
+issue.
+
+## Changelog
+
+- Added support for replacing a comment. It works by adding an invisible tag into the comment, which allows for
```
Comment (piyush-garg) [inline task/github-add-comment/0.6/README.md:8]: This changelog is not valid I guess for this

Comment (frerikandriessen) [reply task/github-add-comment/0.6/README.md:8]: Good point, it's not. Should we even have a changelog? I'd say you can 'easily' find this info with git

**Comment by frerikandriessen** [comment]
@piyush-garg could you give it another look?

**Comment by frerikandriessen** [comment]
@chmouel could you take a look?

**Comment by chmouel** [comment]
<details><summary>Diff between version 0.5 and 0.6</summary>

```diff
diff -urN task/github-add-comment/0.5/github-add-comment.yaml task/github-add-comment/0.6/github-add-comment.yaml
--- task/github-add-comment/0.5/github-add-comment.yaml	2021-11-17 10:45:41.182272845 +0100
+++ task/github-add-comment/0.6/github-add-comment.yaml	2022-01-12 15:52:24.022174639 +0100
@@ -4,7 +4,7 @@
 metadata:
   name: github-add-comment
   labels:
-    app.kubernetes.io/version: "0.5"
+    app.kubernetes.io/version: "0.6"
   annotations:
     tekton.dev/categories: Git
     tekton.dev/pipelines.minVersion: "0.17.0"
@@ -66,6 +66,12 @@
       type: string
       default: token
 
+    - name: AUTH_TYPE
+      description: |
+        The type of authentication to use. You could use the less secure "Basic" for example
+      type: string
+      default: Bearer
+
     - name: COMMENT_TAG
       description: |
         An invisible tag to be added into the comment. The tag is made
@@ -100,6 +106,8 @@
         import sys
         import urllib.parse
 
+        authHeader = "$(params.AUTH_TYPE) " + os.environ["GITHUBTOKEN"]
+
         split_url = urllib.parse.urlparse(
             "$(params.REQUEST_URL)").path.split("/")
 
@@ -139,7 +147,7 @@
             api_url + "/comments",
             headers={
                 "User-Agent": "TektonCD, the peaceful cat",
-                "Authorization": "Bearer " + os.environ["GITHUBTOKEN"],
+                "Authorization": authHeader,
             })
 
           resp = conn.getresponse()
@@ -173,7 +181,7 @@
             body=json.dumps(data),
             headers={
                 "User-Agent": "TektonCD, the peaceful cat",
-                "Authorization": "Bearer " + os.environ["GITHUBTOKEN"],
+                "Authorization": authHeader,
             })
         resp = conn.getresponse()
         if not str(resp.status).startswith("2"):
diff -urN task/github-add-comment/0.5/README.md task/github-add-comment/0.6/README.md
--- task/github-add-comment/0.5/README.md	2021-11-17 10:45:41.182272845 +0100
+++ task/github-add-comment/0.6/README.md	2022-01-12 15:52:24.022174639 +0100
@@ -3,15 +3,10 @@
 The `github-add-comment` task let you add a comment to a pull request or an
 issue.
 
-## Changelog
-
-- Added support for replacing a comment. It works by adding an invisible tag into the comment, which allows for
-  later identification and update of the comment. It introduces two new parameters: `COMMENT_TAG` and `REPLACE`.
-
 ## Install the Task
 
 ```
-kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/github-add-comment/0.5/github-add-comment.yaml
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/github-add-comment/0.6/github-add-comment.yaml
 ```
 
 ## Secrets

**Comment by chmouel** [comment]
the removal of the changelog is intentional?

/lgtm

**Comment by frerikandriessen** [comment]
> the removal of the changelog is intentional?
> 
> /lgtm

Yes, I figured you can 'easily' get this info from the git history, it has a risk of becoming outdated (which is what happened here before @piyush-garg noticed it) and the other tasks that I sampled to double-check did not have a changelog.

**Pull request merged**